encode.go

```diff
@@ -182,10 +183,23 @@ func (e *hjsonEncoder) writeIndent(indent int) {
 	}
 }
 
+func (e *hjsonEncoder) useMarshaler(value reflect.Value, separator string) error {
+	b, err := value.Interface().(json.Marshaler).MarshalJSON()
+	if err != nil {
+		return err
+	}
+	e.WriteString(separator)
+	e.WriteString(string(b))
+	return nil
+}
+
 func (e *hjsonEncoder) str(value reflect.Value, noIndent bool, separator string, isRootObject bool) error {
 
 	// Produce a string from value.
 
+	//first check if value implements Marshaler, and use it
+	m := reflect.TypeOf((*json.Marshaler)(nil)).Elem()
+
```

Comment:
Maybe make this a package-level variable?

Reply:
Done. Also refactored duplicate code https://github.com/mspiegel/hjson-go/pull/1/commits/57a3ee26b3a9f88d372eba2ef840f37c8f0884d9